docs/development/Architecture.md

```diff
@@ -0,0 +1,272 @@
+> **Note**: Please don’t be fanatic with the definition of unitary and integration. There are not always clear boundaries, 
+> but common sense is the best way to identify what needs to be mocked or not.
+
+* `unit`. For unitary tests.
+* `integration`. For integration tests.
+* `resources`. It is also recommended to place here a test_resource_paths.py to have a unique point of access to the actual testing resources.
+* `util`. In case some general utilities need to be developed for being used in the tests, this package is the right place for them.**
+
+### Main Tests Package
+> **Note:**  It is necessary to evaluate the convenience to have this package, but, at least for now, it is important to keep 
+> this tests to guarantee the quality of the code.
+
+At the root level of the Startleft repository (the same as the other modules) is located a general tests package. Unlike 
+the test packages located inside each module, this one is generic to all the Startleft project and contains integration tests 
+that involves logic belonging to all the modules. By definition, this package can only contain integration tests. The boundaries 
+between this kind of tests and the ones developed by QA is hard to define, but it is clear that they are necessary, at least for now. 
+So, **when a new processor or a new important behavior is added to Startleft, we need to create their associated integration 
+tests at project level in this package**.
+
+## Modules Description
+
+---
+### startleft
+This is the main module whose origin is the initial Startleft project that used to contain all the processing logic for 
+the first conversion formats. As part of the modularization process, it was lightened so that it does no longer implements 
+any conversion logic, and it simply exposes the CLI and REST API and makes the connection with the SLPs where the conversion 
+itself takes place.
+
+<u>Main responsibilities</u>: Expose a CLI and a REST API to enable the access to the conversion to OTM logic.
+
+<u>Imported by</u>:  Not imported.
+
+<u>Imports</u>: `_sl_build`, `slp_base`, `sl_util`, `otm`. It also imports `slp_cft` and `slp_tf` to support the 
+search function, 
+whose future is under evaluation.
+
+### sl_util
+This is currently the only module that contains common utilities for the rest of the modules of Startleft. The `sl` prefix 
+means that it is not intended to be a very generic module like the Apache Commons in Java, but a module where place useful 
+functions for the Startleft scope (including some OTM related utilities, for example). However, **it is not a mixed bag to 
+place common logic related, for instance, with the SLPs**. Business logic or domain classes should fit one of the other 
+modules or, if not, maybe a new one should be created.
+
+<u>Main responsibilities</u>: Offer general (not processor specific) utilities related with Startleft.
+
+<u>Imported by</u>:  `startleft`, `slp_base`, SLPs, `otm` (this one should be removed).
```

Comment:
If this can't be removed at the moment, we should specify that it will be removed when the otm module becomes an independent library, if that is the case.

Reply:
Agree, but adding the clarification may be even more confuse. I prefer to simply remove that comment as it does not help so much. Removed in the review branch.